### DISCUSSION TITLE
unsupported to use "export *" in a client boundary

### DISCUSSION THREAD

**Discussion by karthikPattabhiram** [WalletConnect/walletconnect-monorepo#3694] [Developer Support] [answered]
i get this error while setting up web3Modal for my next.js app.

This is what I get 

`./node_modules/wagmi/dist/chains.js
Error: It's currently unsupported to use "export *" in a client boundary. Please use named exports instead.`

**Comment by glitch-txs** [comment]
Hi could you share a MRE?

**Comment by mayjer** [comment, marked as answer]
I ran into this exact error when upgrading to Wagmi `1.x`. The `wagmi/chains` re-exports `viem/chains`. If you look at this file there is no longer an array of all chains, only named exports for each chain.

I had this code in my wagmi config file:
```ts
import * as all chains from "wagmi/chains"
```



* Next complains about this when being imported from a client component, so I made my own explicit list of chains:
```ts
import { type Chain } from "wagmi";
import {  bsc,  bscTestnet,  goerli,  hardhat,  localhost,  mainnet, sepolia } from "viem/chains";

const chainList = [  mainnet,  sepolia,  goerli,  bsc,  bscTestnet,  hardhat,  localhost ];
```

hope that helps!

EDIT:
just changing the import to `viem/chains` also works... it's the `export *` in `wagmi/chains` that Next is actually complaining about

```ts
import * as allChains from "viem/chains";
```
